torch/_dynamo/variables/builder.py

```diff
@@ -1498,6 +1500,135 @@ def wrap_symint(self, value):
+                f"Dynamo attempts to add additional input during export: value={wrapped_value}, source={self.get_source()}"
+            )
+        fake_tensor_value = None
+        example_value = unspec_var.proxy.node.meta["example_value"]
+        assert is_fake(example_value)
+
+        fake_tensor_value = example_value
+        assert fake_tensor_value.fake_mode is self.tx.fake_mode, (
+            f"fake mode ({fake_tensor_value.fake_mode}) from fake tensor metadata doesn't match mode"
+            "({self.tx.fake_mode}) from InstructionTranslator"
+        )
+
+        # There's something a bit incoherent about pass_arg_as_tensor,
+        # specifically regarding sources.
+        #
+        # Specifically, suppose we have "x: float" local argument.  We
+        # eventually end up with an UnspecializedPythonVariable denoting
+        # torch.as_tensor(x)... but it's source is still L['x'] (which if you
+        # accessed it directly is a float!)  So you gotta be careful when
+        # setting up your guards, because it's still going to be a float at
+        # this point, the conversion happens only precisely at the point we're
+        # actually calling the FX graph.  This happens to be what we want for
+        # shape guard generation, but it's kind of unintuitive.
+        proxy.node.meta["grapharg"] = GraphArg(
+            self.get_source(),
+            wrapped_value,
+            pass_arg_as_tensor=True,
+            fake_tensor=fake_tensor_value,
+            is_tensor=False,
+            example_strong_ref=wrapped_value,
+        )
+
+        # OK, now the crazy sauce.  We want to generate a SymNodeVariable to
+        # do the rest of our tracing, doing the equivalent of an item() call.
+        # But we don't /actually/ want to do an item() call, because that will
+        # give us an unbacked SymFloat, but this is really a backed SymFloat.
+
+        item_proxy = self.tx.output.create_proxy(
+            "call_method",
+            "item",
+            (proxy,),
+            {},
+        )
+        # Do NOT do conventional fake tensor prop
+
+        shape_env = self.tx.output.shape_env
+        item_symbol = shape_env.create_unspecified_symbol(
+            value,
+            # !!! Interesting !!!
+            source=self.source,
+            # If we put in a Tensor input, definitely dynamic (if you wanted
+            # it to be static, gotta bail out earlier)
+            dynamic_dim=DimDynamic.DYNAMIC,
+        )
+        item_example_value = shape_env.create_symfloatnode(
+            item_symbol, hint=value, source=self.source
+        )
+        set_example_value(item_proxy.node, item_example_value)
+
+        self.tx.output.tracked_fakes.append(
+            TrackedFake(item_example_value, self.source, None)
+        )
+
+        item_unspec_var = SymNodeVariable(
+            item_proxy,
+            item_example_value,
+            source=self.get_source(),  # !!! Interesting !!!
```

Comment:
can you explain why this is odd?

Reply:
It's odd for the reason stated above, which is that normally if you do compute on a Variable (the compute in this case being an item() call), you end up with a new variable that doesn't have source, but in this case, we can still put a source on it.

Reply:
ah, fair play. Can you leave a teeny tiny comment?